bench/wtperf/wtperf.c

```diff
@@ -445,6 +474,7 @@ worker(void *arg)
 	}
 
 	key_buf = thread->key_buf;
+	truncate_buf = malloc(cfg->key_sz);
```

Comment:
I don't see this getting freed anywhere. Could you use the ```THREAD->key_buf``` instead?

This would show up as a memory leak. It's worth getting into the habit of testing any changes by running in valgrind before pushing changes - that will uncover memory leaks.